src/Symfony/Component/Security/Http/Attribute/IsGranted.php

```diff
@@ -24,18 +26,37 @@
 final class IsGranted
 {
     /**
-     * @param string|Expression            $attribute     The attribute that will be checked against a given authentication token and optional subject
-     * @param array|string|Expression|null $subject       An optional subject - e.g. the current object being voted on
-     * @param string|null                  $message       A custom message when access is not granted
-     * @param int|null                     $statusCode    If set, will throw HttpKernel's HttpException with the given $statusCode; if null, Security\Core's AccessDeniedException will be used
-     * @param int|null                     $exceptionCode If set, will add the exception code to thrown exception
+     * @param string|Expression|null                                                           $attribute     The attribute that will be checked against a given authentication token and optional subject, or a callable that will be called to determine access
+     * @param array|string|Expression|\Closure(Request $request, array $arguments): mixed|null $subject       An optional subject - e.g. the current object being voted on
+     * @param string|null                                                                      $message       A custom message when access is not granted
+     * @param int|null                                                                         $statusCode    If set, will throw HttpKernel's HttpException with the given $statusCode; if null, Security\Core's AccessDeniedException will be used
+     * @param int|null                                                                         $exceptionCode If set, will add the exception code to thrown exception
+     * @param \Closure(AuthorizationCheckerInterface, mixed $subject): bool|null               $callable      A callable that will be called to determine access
      */
     public function __construct(
-        public string|Expression $attribute,
-        public array|string|Expression|null $subject = null,
+        public string|Expression|null $attribute = null,
+        public array|string|Expression|\Closure|null $subject = null,
         public ?string $message = null,
         public ?int $statusCode = null,
         public ?int $exceptionCode = null,
+        public ?\Closure $callable = null,
```

Comment:
why is it a separate argument ? It is the attribute being voted on.

Reply:
Removed, now `$attribute` holds the type